frontend/public/actions/dashboards.ts

```diff
@@ -0,0 +1,95 @@
+import { ActionType as Action, action } from 'typesafe-actions';
+import { Dispatch } from 'react-redux';
+
+import { coFetchJSON } from '../co-fetch';
+import { k8sBasePath } from '../module/k8s/k8s';
+import { RESULTS_TYPE, isWatchActive } from '../reducers/dashboards';
+import { RootState } from '../redux';
+
+export enum ActionType {
+  StopWatch = 'stopWatch',
+  UpdateResult = 'updateResult',
+  ActivateWatch = 'activateWatch',
+  UpdateWatchTimeout = 'updateWatchTimeout',
+  UpdateWatchInFlight = 'updateWatchInFlight',
+}
+
+const prometheusBaseURL = window.SERVER_FLAGS.prometheusBaseURL;
+
+const REFRESH_TIMEOUT = 5000;
+
+export const stopWatch = (type: RESULTS_TYPE, key: string) => action(ActionType.StopWatch, { type, key });
+export const updateResult = (type: RESULTS_TYPE, key: string, result) => action(ActionType.UpdateResult, { type, key, result });
+export const activateWatch = (type: RESULTS_TYPE, key: string) => action(ActionType.ActivateWatch, { type, key });
+export const updateWatchTimeout = (type: RESULTS_TYPE, key: string, timeout) => action(ActionType.UpdateWatchTimeout, { type, key, timeout});
+export const updateWatchInFlight = (type: RESULTS_TYPE, key: string, inFlight: boolean) => action(ActionType.UpdateWatchInFlight, { type, key, inFlight });
+
+const dashboardsActions = {
```

Comment:
Please add `__tests__/actions/dashboards.spec.ts`.

Reply:
added tests for dashboard actions and reducers